lib/generators/generators.py

```diff
@@ -78,8 +164,8 @@ def load_generator(module: str, context: Optional[Mapping] = None) -> Generator:
 
     try:
         instance_id = hash_nc(f"{module_name}{str(context)}", 8)
-        if instance_id not in _generator_instances:
-            _generator_instances[module_name][instance_id] = _loaded_generators[module_name](context)
+        if instance_id not in _generator_instances[module_name]:
+            _generator_instances[module_name][instance_id] = _loaded_generators[module_name](context, instance_id)
```

Comment:
Question: 2 generators w/ same module and same context are considered the same? I.e. I have 2 RRgenerators with the same sequence in 2 different places in the config. Will they implement the same cycle (I call one and rotates the elements also for the other)? OK to be that way but should be specified in docstring and documentation so operators can ass an element in the context if they desire something different.